report/mapping_functions.go

```diff
@@ -25,6 +26,17 @@ type MappedNode struct {
 // rendered topology.
 type MapFunc func(string, NodeMetadata, bool) (MappedNode, bool)
 
+// PseudoFunc is anything which can take an arbitrary node id and produce
+// a pseudo node for the given id, in the form of a node ID and a
+// human-readable major and minor labels.
+//
+// First input param is the src id (before mapping), second is the src rendered node,
+// third is the dst id, fourth is grouped.
+//
+// If the final output parameter is false, the node shall be omitted from the
+// rendered topology.
+type PseudoFunc func(string, RenderableNode, string, bool) (MappedNode, bool)
```

Comment:
It doesn't strictly map the renderable nodes to a pseudo node, but a node id to pseudo node (as there is no renderable node by definition).  I've made the role of srcNode more obvious.